src/olympia/scanners/models.py

```diff
@@ -81,7 +81,10 @@ def extract_rule_names(self):
     def save(self, *args, **kwargs):
         rule_model = self._meta.get_field('matched_rules').related_model
         matched_rules = rule_model.objects.filter(
-            scanner=self.scanner, name__in=self.extract_rule_names()
+            scanner=self.scanner,
+            name__in=self.extract_rule_names(),
+            # See: https://github.com/mozilla/addons-server/issues/13143
+            is_active=True,
         )
         self.has_matches = bool(matched_rules)
         # Save the instance first...
```

Comment:
I'm aware this isn't new functionality in this commit, but wouldn't this add a new `matched_rule` _each_ time a `AbstractScannerResult` instance is saved?  So the second time the instance is saved it'd create a new `ScannerMatch` record, even though it already exists - or is there some de-duplication going on elsewhere?

Reply:
Good question, I assume this is handled _via_ the ORM. I'll double-check and file an issue if needed. /cc @diox

Reply:
There's no built-in duplication, afaik.  Either there will be duplicate many-to-many links on subsequent saves, or a database error, depending on what constraints there are in place around uniqueness.

Reply:
There is, in fact. Django does this for us. I've checked it when we introduced it, because I also thought this was wrong, but it's fine :) It's documented in https://docs.djangoproject.com/en/3.0/topics/db/examples/many_to_many/:
> Adding a second time is OK, it will not duplicate the relation

You can test it locally, it doesn't create additional matches for dupes.

Reply:
Thanks for confirming. That sounds like something an ORM should do (otherwise, what's the point? :P).

I think we only save the scanner results once anyway (we don't provide tools to update these models) but that was a good point to double check.